server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -399,6 +399,16 @@ public Iterator<Setting<?>> settings() {
 
     public static final ByteSizeValue DEFAULT_CHUNK_SIZE = new ByteSizeValue(512, ByteSizeUnit.KB);
 
+    /**
+     * The maximum allowable size, in bytes, for buffering source documents during recovery.
+     */
+    public static final Setting<ByteSizeValue> INDICES_RECOVERY_CHUNK_SIZE = Setting.byteSizeSetting(
```

Comment:
The method `setChunkSize` has some validation if it's negative, we need to validate that in the setting too.

Reply:
Added the validation in https://github.com/elastic/elasticsearch/pull/114618/commits/c8625d88cf9962d53288e9168127755bc8fc9333